actions/global_actions.jsx

```diff
@@ -25,200 +26,192 @@ import {closeRightHandSide, closeMenu as closeRhsMenu, updateRhsState} from 'act
+        function switchToChannel(chan) {
+            const state = getState();
+            const userId = getCurrentUserId(state);
+            const teamId = chan.team_id || getCurrentTeamId(state);
+            const isRHSOpened = getIsRhsOpen(state);
+            const isPinnedPostsShowing = getRhsState(state) === RHSStates.PIN;
+            const member = getMyChannelMember(state, chan.id);
+
+            dispatch(getChannelStats(chan.id));
+
+            if (chan.delete_at === 0) {
+                const penultimate = LocalStorageStore.getPreviousChannelName(userId, teamId);
+                if (penultimate !== chan.name) {
+                    LocalStorageStore.setPenultimateChannelName(userId, teamId, penultimate);
+                    LocalStorageStore.setPreviousChannelName(userId, teamId, chan.name);
+                }
             }
-        );
-    } else {
-        switchToChannel(channel);
-    }
-}
 
-export function emitCloseRightHandSide() {
-    dispatch(closeRightHandSide());
-}
+            // When switching to a different channel if the pinned posts is showing
+            // Update the RHS state to reflect the pinned post of the selected channel
+            if (isRHSOpened && isPinnedPostsShowing) {
+                dispatch(updateRhsState(RHSStates.PIN, chan.id));
+            }
 
-export function toggleShortcutsModal() {
-    AppDispatcher.handleViewAction({
-        type: ActionTypes.TOGGLE_SHORTCUTS_MODAL,
-        value: true,
-    });
-}
+            loadProfilesForSidebar();
+
+            dispatch(batchActions([{
+                type: ChannelTypes.SELECT_CHANNEL,
+                data: chan.id,
+            }, {
+                type: ActionTypes.SELECT_CHANNEL_WITH_MEMBER,
+                data: chan.id,
+                channel: chan,
+                member: member || {},
+            }]));
+        }
 
-export function showChannelPurposeUpdateModal(channel) {
-    AppDispatcher.handleViewAction({
-        type: ActionTypes.TOGGLE_CHANNEL_PURPOSE_UPDATE_MODAL,
-        value: true,
-        channel,
-    });
+        if (channel.fake) {
+            userVisitedFakeChannel(
+                channel,
+                (data) => {
+                    switchToChannel(data);
+                },
+                () => {
+                    browserHistory.push('/' + this.state.currentTeam.name);
+                }
+            );
+        } else {
+            switchToChannel(channel);
+        }
+    };
 }
 
-export function showChannelNameUpdateModal(channel) {
-    AppDispatcher.handleViewAction({
-        type: ActionTypes.TOGGLE_CHANNEL_NAME_UPDATE_MODAL,
-        value: true,
-        channel,
-    });
-}
+export function showShortcutsModal() {
+    return (dispatch) => {
+        const shortcutsModalData = {
+            modalId: ModalIdentifiers.SHORTCUTS,
+            dialogType: ShortcutsModal,
+            dialogProps: {
+                isMac: Utils.isMac(),
```

Comment:
I was actually thinking the same thing, but I passed it in because it was originally passed into the component and wanted to minimize changes to ShortcutsModal.

Reply:
I mean in the modal itself. This does not have to be a prop as it does not change when app is running.